File: 908-middle-of-the-linked-list/middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode s=head;
        ListNode d=head;
        while(d!=null && d.next!=null){
            s=s.next;
            d=d.next.next;
             }
        return s;
    }
}
        
    

File: 908-middle-of-the-linked-list/middle-of-the-linked-list.py
# Definition for singly-linked list.
class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution(object):
    def middleNode(self, head):
        fast = head
        slow = head
        while fast and fast.next:  # Check both fast and fast.next
            fast = fast.next.next
            slow = slow.next
        head=slow
        return head  # Return the value of the middle node

        
        